### DISCUSSION TITLE
Is it possible to escape a user input for a LIKE operator?

### DISCUSSION THREAD

**Discussion by muturgan** [sidorares/node-mysql2#1494] [Q&A] [answered]
I need a some sort of query like this:
```js
`SELECT * FROM my_table WHERE name LIKE '%${ userInput }%'`
```
If I will use a ? placeholder and pass a second param to the query with a userInput then the generated sql will be broken
```js
`SELECT * FROM my_table WHERE name LIKE '%'Mike'%'`
```
Is it possible to escape a user input for a LIKE operator with a built in mysql2 tools?

**Comment by dougwilson** [comment]


**Reply by muturgan** [reply]
As I can see a user input is still unsafe

**Comment by muturgan** [comment, marked as answer]
I like an answer from SO - using position instead of like:
```
`SELECT * FROM my_table WHERE POSITION(? IN name) > 0',
[ userInput ]
```
